libraries/AP_UAVCAN/AP_UAVCAN.cpp

```diff
@@ -497,6 +545,14 @@ bool AP_UAVCAN::try_init(void)
                         return false;
                     }
 
+                    uavcan::Subscriber<uavcan::protocol::NodeStatus> *nodestatus_st;
+                    nodestatus_st = new uavcan::Subscriber<uavcan::protocol::NodeStatus>(*node);
```

Comment:
check for alloc failure?

Reply:
yes, yes, the whole AP_UAVCAN library needs a total re-think. The existing code is just "more of the same". In a different PR we need to restructure all of this.